mindmeld/core.py

```diff
@@ -747,6 +747,38 @@ def __repr__(self):
         return "<{} {!r} ({!r})>".format(self.__class__.__name__, text, self.type)
 
 
+class FormEntity:
+    """A form entity is used for defining custom objects for the entity form used in
+    AutoEntityFilling (slot-filling).
+
+    Attributes:
+        entity (str): Entity name
+        role (str): The role of the entity
+        responses(list): NLR for prompting the user for missing entities
+        value (str): The resolved value of the entity
+        default_eval(bool): Use system validation (default: True)
+        hints(list): Developer defined list of keywords to verify the user input against
+        custom_eval(func): custom validation function
+    """
+    def __init__(
+        self,
+        entity,
+        role=None,
+        responses=None,
```

Comment:
We should validate:
1) entity name is not None
2) there is at least one Response

Alternatively, for 2) we can also set a default value for Response as well

Reply:
2 is already done (default response).

1 is handled currently by exiting the turn.